src/Core/Worker.php

```diff
@@ -1140,6 +1149,32 @@ private static function runCron()
 
 		// Cleaning dead processes
 		self::killStaleWorkers();
+
+		// Remove old entries from the workerqueue
+		self::cleanWorkerQueue();
+	}
+
+	/**
+	 * Remove old entries from the workerqueue
+	 *
+	 * @return void
+	 */
+	private static function cleanWorkerQueue()
+	{
+		DBA::delete('workerqueue', ["`done` AND `executed` < ?", DateTimeFormat::utc('now - 1 hour')]);
```

Comment:
Why so short a delay? Why not directly deleting the row when execution is done?

Reply:
It is needed I guess for the statistics to keep them for some minutes. I just copied that part from the worker task.